src/api/recipes.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from "@tanstack/react-query";
+import { DocumentData, getDocs } from "firebase/firestore";
+import { Recipe } from "../types/RecipeTypes";
+import { FirebaseError } from "firebase/app";
+import { QueryKeys } from "../enums/QueryKeys";
+import { Collections } from "../enums/Collections";
+import { createCollection } from "../utils/recipeUtils";
+
+export function useFetchRecipes() {
+  return useQuery<Recipe[], FirebaseError>({
+    queryKey: [QueryKeys.recipesData],
+    queryFn: async () => {
+      const recipeCollection = createCollection<Recipe>(Collections.recipes);
```

Comment:
you can export it in main firebase file so you don't have to create it everytime you need it